### ISSUE TITLE
AttributeError: 'Frame' object has no attribute 'transmitters'

### ISSUE THREAD

**Issue by VincieD** [ebroecker/canmatrix#280]
While trying to import CAN arxml I'm facing following problem:

AttributeError: 'Frame' object has no attribute 'transmitters'

    arxml1Db = next(iter(canmatrix.formats.loadp(file).values()))

    pgn, id = list_pgn(db=arxml1Db)   
    for ecu in arxml1Db.boardUnits:
        print(ecu.name)
        
    for frame in arxml1Db.frames:
        print("- {}".format(frame.name))
       
    for idx, pgnx in zip(id, pgn):
        Fr = arxml1Db.frameById(idx)
        for sig in Fr.signals:
            print(sig.name)
        for tran in Fr.transmitters:
            print(tran.name)

I can see in canmatrix.py that class Frame should have a list of transmitters. Moreover it is used in functions like updateEcuList(self). 

Can anybody tell me why it is in my case not working, or show me better approach how to iterate through whole C-Matrix (ECUs, Frames and recieved/transmitted Signals)

Thank you,
Vaclav

**Comment by ebroecker** [comment]
maybe your arxml does not have transmitters defined?
If you provide your ARXML (maybe as PN) I could have a look on it.
Eduard

**Comment by VincieD** [comment]
I can't provide you the arxml. Maybe we could see it on some screenshots? I can open the arxml with Autosar Explorer how you can see on the screenshot bellow. Here are defined senders and receivers and i quickly checked it and all Frames have at least one sender and one receiver.
<img width="220" alt="2019-02-04 16_14_21-" src="https://user-images.githubusercontent.com/10423833/52217400-a632d500-2898-11e9-82bc-71c2fdc8a149.png">
Thank you!

**Comment by rutgervandenberg** [comment]
Most likely updating your version will fix this. It was changed in https://github.com/ebroecker/canmatrix/commit/81a40ad621d1778b6284faed10f6c6b906b23cda

I noticed just now because my CI builds were failing on this change.

**Comment by VincieD** [comment]
Thank you! I'm facing a problem with the latest version. It doesnt want to open my arxml at all.
Funny thing - version 0.6 works fine only for tranciever but return empty list of receiver. Do you face the same problem? Since my arxml contains receivers, it is not obviously a problem o arxml but somewhere between chair and keyboard :)

**Comment by ebroecker** [comment]
@VincieD 
Just for understanding you right:

0.6 works but 0.7 does not work?

**Comment by VincieD** [comment]
0.7 Version doesn't open the arxml at all:

> File "C:\Users\divis\Desktop\FirstApp\scripts\FlowChart.py", line 106, in readArxml
  **arxml1Db = next(iter(canmatrix.formats.loadp(file).values()))**
  File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\formats.py", line 64, in loadp
    return load(fileObject, importType, key, flatImport, **options)
  File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\formats.py", line 74, in load
    dbs = moduleInstance.load(fileObject, **options)
  File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\arxml.py", line 1612, in load
    frameObject = getFrame(frameTrig,searchPoint,multiplexTranslation,ns, float_factory)
  File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\arxml.py", line 1369, in getFrame
    getSignals(pdusigmapping, newFrame, xmlRoot, ns, None, float_factory)
  File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\arxml.py", line 1033, in getSignals
    baseType = arGetChild(isignal,"BASE-TYPE", xmlRoot, ns)
  File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\arxml.py", line 928, in arGetChild
    ret = getArPath(xmlRoot, ret.text, namespace)
  File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\arxml.py", line 897, in getArPath
    **baseElement = tree.xpath(xbasePath, namespaces=namespaceMap)[0]
IndexError: list index out of range**

And with the 0.6 version an empty list of receivers and half full transmitters is received:


        file = "lokal.arxml"

        arxml1Db = next(iter(canmatrix.formats.loadp(file).values()))

        pgn, id = self.list_pgn(db=arxml1Db)

        for frame in arxml1Db.frames:
            self.frames.append(frame.name)
  
        for ecu in arxml1Db.boardUnits:
            self.ecus.append(ecu.name)  
        
        for idx, pgnx in zip(id, pgn):
            Fr = arxml1Db.frameById(idx)
            for sig in Fr.signals:
                self.signals.append(sig.name)
            
            for tra in Fr.transmitter:
                self.fun.set_key(self.transmitters, tra, Fr.name)

            for rec in Fr.receiver:
                self.fun.set_key(self.receivers, rec, Fr.name)

Thank you very much for your support!

**Comment by ebroecker** [comment]
please try:
```python
arxml1Db = next(iter(canmatrix.formats.loadp(file, arxmlUseXpath = False).values()))
```

**Comment by VincieD** [comment]
1) after doing as you said, i found out following:

> File "C:\Python36\lib\site-packages\canmatrix-0+unknown-py3.6.egg\canmatrix\canmatrix.py", line 892, in signals_to_bytes
    least = self.size * 8 - signal.startBit
TypeError: unsupported operand type(s) for -: 'int' and 'str'

2) simple cast worked:
`least = self.size * 8 - int(signal.startBit)`

3) after that: 

> for rec in Fr.receivers:
AttributeError: 'Frame' object has no attribute 'receivers'

4) so i've changed it to 'receiver'.  But the list is empty

**Comment by ebroecker** [comment]
the TypeError could be fixed with c8cc7e9

Every frame should have receiver...
```python
>>> import canmatrix
>>> f = canmatrix.Frame("test")
>>> f.receiver
[]
>>> 
```

**Comment by ebroecker** [comment]
should be fixed
